### PR TITLE
Add upgrading doc for 0.6 from 0.4

### PR DESCRIPTION
**Pull request by agunde406**
This document contains important notes on breaking
features and required updates.

Signed-off-by: Andrea Gunderson <agunde@bitwise.io>

### CODE REVIEW

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+If using `--peers` on `splinterd` to connect to other nodes without a circuit,
+Challenge Authorization will be used as default also. You can change the key
+that it is used for this authorization with ` --peering-key`. It must be a key
+stored in SPLINTER_CONFIG_DIR/keys/. If you are trying to connect to to a 0.4
+node and need to use trust, append `+trust` to the protocol prefix.
+
+```
+$ splinterd
+     --peering-key other_key \
```
Comment (vaporos) [inline releases/0.6/upgrading/splinter-v0.6-from-v0.4.md:293]: Additional space indent here and missing \ on the line above.

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    --peers tcp://127.0.0.1:8000 \
+    --peers tcp+trust://127.0.0.1:8001 \
+    . . .
+```
+
+## Certain updates to circuits are not compatible with 0-4
```
Comment (vaporos) [inline releases/0.6/upgrading/splinter-v0.6-from-v0.4.md:299]: Change 0-4 to 0.4. "0-4" refers to the git branch, not the version.

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md (hunk 1 above)
Comment (agunde406) [reply releases/0.6/upgrading/splinter-v0.6-from-v0.4.md]: fixed

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md (hunk 2 above)
Comment (agunde406) [reply releases/0.6/upgrading/splinter-v0.6-from-v0.4.md]: fixed

**Review by vaporos — DISMISSED** [review]

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+verification.
+
+In 0.6, the default authorization type has been updated to Challenge
+Authorization. Challenge Authorization requires a node’s ID to be tied to a
+public key/private key pair and a node must prove they have access to that key
+by signing a random nonce, providing the resulting signature and their public
+key.
+
+This requires that a `splinterd` has its own system key that will be used for
+identification. The key must be stored in the SPLINTER_CONFIG_DIR/keys/. If
```
Comment (rbuysse) [inline releases/0.6/upgrading/splinter-v0.6-from-v0.4.md:262]: stored at? in the `SPLINTER_CONFIG_DIR/keys/` directory?

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+multiple keys exist in this location, all of them will be loaded into
+`splinterd` and can be used for Challenge Authorization. To create this key run:
+
+```
+$ splinter keygen --system
+```
+
+When creating a circuit with the CLI, the public keys for the other members must
+be known at provided. Use `--node-public-key` for each node:
```
Comment (rbuysse) [inline releases/0.6/upgrading/splinter-v0.6-from-v0.4.md:271]: s/known at/known and/

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    --node-public-key n8198::02bf74d9263327a571763c6557f50d7995bf3dec86387fc8e5f9f75a74b15919a4 \
+    . . .
+```
+
+To continue using Trust authorization, set `--auth-type trust` when proposing
+the circuit.
+
+If using `--peers` on `splinterd` to connect to other nodes without a circuit,
+Challenge Authorization will be used as default also. You can change the key
+that it is used for this authorization with ` --peering-key`. It must be a key
```
Comment (rbuysse) [inline releases/0.6/upgrading/splinter-v0.6-from-v0.4.md:287]: This formatting is broken
Screen Shot 2021-12-15 at 1 12 06 PM

**Review by rbuysse — CHANGES_REQUESTED** [review]

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md (hunk 3 above)
Comment (agunde406) [reply releases/0.6/upgrading/splinter-v0.6-from-v0.4.md]: fixed (went with directory)

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md (hunk 4 above)
Comment (agunde406) [reply releases/0.6/upgrading/splinter-v0.6-from-v0.4.md]: fixed

File: releases/0.6/upgrading/splinter-v0.6-from-v0.4.md (hunk 5 above)
Comment (agunde406) [reply releases/0.6/upgrading/splinter-v0.6-from-v0.4.md]: fixed

**Review by vaporos — APPROVED** [review]

**Review by rbuysse — APPROVED** [review]

**Pull request merged**